platform/configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/AdminOperation.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.management.MalformedObjectNameException;
+
+import org.codice.ddf.admin.configurator.ConfiguratorException;
+import org.codice.ddf.admin.configurator.Operation;
+import org.codice.ddf.ui.admin.api.ConfigurationAdminMBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Transactional handler for persisting bundle configuration file changes.
+ */
+public class AdminOperation implements OperationBase, Operation<Void, Map<String, Object>> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AdminOperation.class);
+
+    private final String pid;
+
+    private final Map<String, Object> configs;
+
+    private final boolean keepIgnored;
+
+    private final ConfigurationAdminMBean cfgAdmMbean;
+
+    private Map<String, Object> currentProperties;
+
+    private AdminOperation(String pid, Map<String, Object> configs, boolean keepIgnored,
+            ConfigurationAdminMBean cfgAdmMbean) {
+        this.pid = pid;
+        this.configs = new HashMap<>(configs);
+        this.keepIgnored = keepIgnored;
+        this.cfgAdmMbean = cfgAdmMbean;
+
+        try {
+            currentProperties = cfgAdmMbean.getProperties(pid);
+        } catch (IOException e) {
+            LOGGER.debug("Error getting current configuration for pid {}", pid, e);
+            throw new ConfiguratorException("Internal error");
```

Comment:
@paouelle Generally, we throw a generic exception if it is one we expect to be returned to a client for security reasons. I'm not sure if this is the case here. We usually expect the generic messages to be returned (with the actuals logged) from wherever the actual endpoint is and then flow stack traces up from below that.